client/state/oauth2-clients/test/reducer.js

```diff
@@ -111,20 +80,4 @@ describe( 'reducer', () => {
 			},
 		} );
 	} );
-
-	test( 'should not persist state', () => {
```

Comment:
This seems useful to ensure. Perhaps testing it with a solo `combineReducers`?

Reply:
That would just duplicate the unit tests of `combineReducers`.